src/GridTurbulence.cpp.in

```diff
@@ -365,39 +456,45 @@ void initTurbulenceWithBendover(ref_ptr<VectorGrid> grid, double Brms, double lM
 
 	Vector3d spacing = grid->getSpacing();
 	size_t n = grid->getNx(); // size of array
-	size_t n2 = (size_t) floor(n / 2) + 1; // size array in z-direction in configuration space
+	size_t n2 = (size_t) std::floor(n / 2) + 1; // size array in z-direction in configuration space
 
 	// arrays to hold the complex vector components of the B(k)-field
-	fftwf_complex *Bkx, *Bky, *Bkz;
-	Bkx = (fftwf_complex*) fftwf_malloc(sizeof(fftwf_complex) * n * n * n2);
-	Bky = (fftwf_complex*) fftwf_malloc(sizeof(fftwf_complex) * n * n * n2);
-	Bkz = (fftwf_complex*) fftwf_malloc(sizeof(fftwf_complex) * n * n * n2);
+	@FFTW_PREFIX@_complex *Bkx, *Bky, *Bkz;
+	Bkx = (@FFTW_PREFIX@_complex*) @FFTW_PREFIX@_malloc(sizeof(@FFTW_PREFIX@_complex) * n * n * n2);
+	Bky = (@FFTW_PREFIX@_complex*) @FFTW_PREFIX@_malloc(sizeof(@FFTW_PREFIX@_complex) * n * n * n2);
+	Bkz = (@FFTW_PREFIX@_complex*) @FFTW_PREFIX@_malloc(sizeof(@FFTW_PREFIX@_complex) * n * n * n2);
 
 	Random random;
 	if (seed != 0)
 		random.seed(seed); // use given seed
 
 	// calculate the n possible discrete wave numbers
+	// k in [-n/2, n/2>
 	double K[n];
 	for (int i = 0; i < n; i++)
-		K[i] = ((double) i / n - i / (n / 2));
+		K[i] = (static_cast<double>(i) - i / (n / 2) * n);
```

Comment:
These lines are not equivalent - Should probably read (static_cast<double>(i) / n - i / (n / 2));